runtime.go:66 panic, nullpointer exception

**Kong Ingress controller version**: 0.0.3


**Kubernetes version** (use `kubectl version`):

```
kubectl version
Client Version: version.Info{Major:"1", Minor:"9", GitVersion:"v1.9.0", GitCommit:"925c127ec6b946659ad0fd596fa959be43f0cc05", GitTreeState:"clean", BuildDate:"2017-12-16T03:15:38Z", GoVersion:"go1.9.2", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"10", GitVersion:"v1.10.0", GitCommit:"fc32d2f3698e36b93322a3465f63a14e9f0eaead", GitTreeState:"clean", BuildDate:"2018-03-26T16:44:10Z", GoVersion:"go1.9.3", Compiler:"gc", Platform:"linux/amd64"}
```


**Environment**:

- **Cloud provider or hardware configuration**: OSX 10.12.6, Ubuntu 16.04.1
- **OS** (e.g. from /etc/os-release): minikube version: v0.26.1
- **Install tools**: Helm
```
Client: &version.Version{SemVer:"v2.6.1", GitCommit:"bbc1f71dc03afc5f00c6ac84b9308f8ecb4f39ac", GitTreeState:"clean"}
Server: &version.Version{SemVer:"v2.6.1", GitCommit:"bbc1f71dc03afc5f00c6ac84b9308f8ecb4f39ac", GitTreeState:"clean"}
```


**What happened**:
```
kubectl logs kong-ingress-controller-9d45b6677-mlcnl -n kong ingress-controller
-------------------------------------------------------------------------------
Kong Ingress controller
  Release:    0.0.3
  Build:      git-51fa668
  Repository: https://github.com/Kong/kubernetes-ingress-controller.git
-------------------------------------------------------------------------------

W0524 12:56:12.856031       6 client_config.go:533] Neither --kubeconfig nor --master was specified.  Using the inClusterConfig.  This might not work.
I0524 12:56:12.856316       6 main.go:204] Creating API client for https://10.96.0.1:443
I0524 12:56:12.910746       6 main.go:248] Running in Kubernetes Cluster version v1.10 (v1.10.0) - git (clean) commit fc32d2f3698e36b93322a3465f63a14e9f0eaead - platform linux/amd64
I0524 12:56:12.930063       6 main.go:94] validated kong/kong-proxy as the default backend
I0524 12:56:13.569503       6 main.go:151] kong version: 0.13.1
I0524 12:56:13.690679       6 run.go:144] starting Ingress controller
I0524 12:56:14.934969       6 event.go:218] Event(v1.ObjectReference{Kind:"Ingress", Namespace:"kong-test", Name:"proxy-from-k8s-to-mockbin", UID:"b05c3e75-5f4d-11e8-9c11-080027339b80", APIVersion:"extensions", ResourceVersion:"1792", FieldPath:""}): type: 'Normal' reason: 'CREATE' Ingress kong-test/proxy-from-k8s-to-mockbin
I0524 12:56:14.935365       6 event.go:218] Event(v1.ObjectReference{Kind:"Ingress", Namespace:"kong-test", Name:"proxy-from-k8s-jwt-to-mockbin", UID:"b05f3af9-5f4d-11e8-9c11-080027339b80", APIVersion:"extensions", ResourceVersion:"1794", FieldPath:""}): type: 'Normal' reason: 'CREATE' Ingress kong-test/proxy-from-k8s-jwt-to-mockbin
I0524 12:56:14.994391       6 leaderelection.go:175] attempting to acquire leader lease  kong/ingress-controller-leader-nginx...
I0524 12:56:15.036140       6 leaderelection.go:184] successfully acquired lease kong/ingress-controller-leader-nginx
I0524 12:56:15.036397       6 status.go:217] new leader elected: kong-ingress-controller-9d45b6677-mlcnl
I0524 12:56:18.330493       6 controller.go:127] syncing Ingress configuration...
E0524 12:56:18.334614       6 runtime.go:66] Observed a panic: "invalid memory address or nil pointer dereference" (runtime error: invalid memory address or nil pointer dereference)
/home/aledbf/go/src/github.com/kong/kubernetes-ingress-controller/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:72
/home/aledbf/go/src/github.com/kong/kubernetes-ingress-controller/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:65
/home/aledbf/go/src/github.com/kong/kubernetes-ingress-controller/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:51
/home/aledbf/.gimme/versions/go1.10.1.linux.amd64/src/runtime/asm_amd64.s:573
/home/aledbf/.gimme/versions/go1.10.1.linux.amd64/src/runtime/panic.go:502
/home/aledbf/.gimme/versions/go1.10.1.linux.amd64/src/runtime/panic.go:63
/home/aledbf/.gimme/versions/go1.10.1.linux.amd64/src/runtime/signal_unix.go:388
/home/aledbf/go/src/github.com/kong/kubernetes-ingress-controller/internal/ingress/controller/kong.go:885
/home/aledbf/go/src/github.com/kong/kubernetes-ingress-controller/internal/ingress/controller/kong.go:77
/home/aledbf/go/src/github.com/kong/kubernetes-ingress-controller/internal/ingress/controller/controller.go:128
/home/aledbf/go/src/github.com/kong/kubernetes-ingress-controller/internal/ingress/controller/run.go:85
/home/aledbf/go/src/github.com/kong/kubernetes-ingress-controller/internal/task/queue.go:112
/home/aledbf/go/src/github.com/kong/kubernetes-ingress-controller/internal/task/queue.go:59
/home/aledbf/go/src/github.com/kong/kubernetes-ingress-controller/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133
/home/aledbf/go/src/github.com/kong/kubernetes-ingress-controller/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:134
/home/aledbf/go/src/github.com/kong/kubernetes-ingress-controller/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88
/home/aledbf/go/src/github.com/kong/kubernetes-ingress-controller/internal/task/queue.go:59
/home/aledbf/.gimme/versions/go1.10.1.linux.amd64/src/runtime/asm_amd64.s:2361
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x10ec7c4]

goroutine 180 [running]:
github.com/kong/kubernetes-ingress-controller/vendor/k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	/home/aledbf/go/src/github.com/kong/kubernetes-ingress-controller/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:58 +0x107
panic(0x12a8fa0, 0x1eb6090)
	/home/aledbf/.gimme/versions/go1.10.1.linux.amd64/src/runtime/panic.go:502 +0x229
github.com/kong/kubernetes-ingress-controller/internal/ingress/controller.(*NGINXController).syncUpstreams(0xc4201c43f0, 0xc4207be100, 0x3, 0x4, 0xc42026f9f0, 0x2, 0x2, 0x1ecece0, 0xc400000000)
	/home/aledbf/go/src/github.com/kong/kubernetes-ingress-controller/internal/ingress/controller/kong.go:885 +0x984
github.com/kong/kubernetes-ingress-controller/internal/ingress/controller.(*NGINXController).OnUpdate(0xc4201c43f0, 0xc4202b9ce0, 0x0, 0x0)
	/home/aledbf/go/src/github.com/kong/kubernetes-ingress-controller/internal/ingress/controller/kong.go:77 +0xc8
github.com/kong/kubernetes-ingress-controller/internal/ingress/controller.(*NGINXController).syncIngress(0xc4201c43f0, 0x1315940, 0xc4203c98c0, 0xc43b610246, 0x88773e8f)
	/home/aledbf/go/src/github.com/kong/kubernetes-ingress-controller/internal/ingress/controller/controller.go:128 +0x2cd
github.com/kong/kubernetes-ingress-controller/internal/ingress/controller.(*NGINXController).(github.com/kong/kubernetes-ingress-controller/internal/ingress/controller.syncIngress)-fm(0x1315940, 0xc4203c98c0, 0xa, 0xc42062bd68)
	/home/aledbf/go/src/github.com/kong/kubernetes-ingress-controller/internal/ingress/controller/run.go:85 +0x3e
github.com/kong/kubernetes-ingress-controller/internal/task.(*Queue).worker(0xc42025fd70)
	/home/aledbf/go/src/github.com/kong/kubernetes-ingress-controller/internal/task/queue.go:112 +0x34a
github.com/kong/kubernetes-ingress-controller/internal/task.(*Queue).(github.com/kong/kubernetes-ingress-controller/internal/task.worker)-fm()
	/home/aledbf/go/src/github.com/kong/kubernetes-ingress-controller/internal/task/queue.go:59 +0x2a
github.com/kong/kubernetes-ingress-controller/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1(0xc4204a17a8)
	/home/aledbf/go/src/github.com/kong/kubernetes-ingress-controller/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133 +0x54
github.com/kong/kubernetes-ingress-controller/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc420365fa8, 0x3b9aca00, 0x0, 0xc42011d601, 0xc4203085a0)
	/home/aledbf/go/src/github.com/kong/kubernetes-ingress-controller/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:134 +0xbd
github.com/kong/kubernetes-ingress-controller/vendor/k8s.io/apimachinery/pkg/util/wait.Until(0xc4204a17a8, 0x3b9aca00, 0xc4203085a0)
	/home/aledbf/go/src/github.com/kong/kubernetes-ingress-controller/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88 +0x4d
github.com/kong/kubernetes-ingress-controller/internal/task.(*Queue).Run(0xc42025fd70, 0x3b9aca00, 0xc4203085a0)
	/home/aledbf/go/src/github.com/kong/kubernetes-ingress-controller/internal/task/queue.go:59 +0x55
created by github.com/kong/kubernetes-ingress-controller/internal/ingress/controller.(*NGINXController).Start
	/home/aledbf/go/src/github.com/kong/kubernetes-ingress-controller/internal/ingress/controller/run.go:152 +0xe6
```

**What you expected to happen**:
Ingress controller not no crash.

**Anything else we need to know**:

Previously a similar error has been reported.

Updated to ingress 0.0.4

```
kubectl logs -n kong kong-ingress-controller-65549cc67d-mxsxr ingress-controller
-------------------------------------------------------------------------------
Kong Ingress controller
  Release:    0.0.4
  Build:      git-d691586
  Repository: https://github.com/Kong/kubernetes-ingress-controller.git
-------------------------------------------------------------------------------

W0524 13:31:52.196652       5 client_config.go:533] Neither --kubeconfig nor --master was specified.  Using the inClusterConfig.  This might not work.
I0524 13:31:52.196885       5 main.go:204] Creating API client for https://10.96.0.1:443
I0524 13:31:52.206282       5 main.go:248] Running in Kubernetes Cluster version v1.10 (v1.10.0) - git (clean) commit fc32d2f3698e36b93322a3465f63a14e9f0eaead - platform linux/amd64
I0524 13:31:52.209872       5 main.go:94] validated kong/kong-proxy as the default backend
I0524 13:31:52.533933       5 main.go:151] kong version: 0.13.1
I0524 13:31:52.588004       5 run.go:144] starting Ingress controller
I0524 13:31:53.700020       5 event.go:218] Event(v1.ObjectReference{Kind:"Ingress", Namespace:"kong-test", Name:"proxy-from-k8s-jwt-to-mockbin", UID:"f8039ad0-5f50-11e8-a248-080027ca920b", APIVersion:"extensions", ResourceVersion:"10596", FieldPath:""}): type: 'Normal' reason: 'CREATE' Ingress kong-test/proxy-from-k8s-jwt-to-mockbin
I0524 13:31:53.700063       5 event.go:218] Event(v1.ObjectReference{Kind:"Ingress", Namespace:"kong-test", Name:"proxy-from-k8s-to-mockbin", UID:"c74ee38b-5f44-11e8-8656-080027ca920b", APIVersion:"extensions", ResourceVersion:"4093", FieldPath:""}): type: 'Normal' reason: 'CREATE' Ingress kong-test/proxy-from-k8s-to-mockbin
I0524 13:31:53.790280       5 leaderelection.go:175] attempting to acquire leader lease  kong/ingress-controller-leader-nginx...
I0524 13:31:53.807461       5 leaderelection.go:184] successfully acquired lease kong/ingress-controller-leader-nginx
I0524 13:31:53.807500       5 status.go:217] new leader elected: kong-ingress-controller-65549cc67d-mxsxr
I0524 13:31:57.124862       5 controller.go:127] syncing Ingress configuration...
E0524 13:31:57.128775       5 runtime.go:66] Observed a panic: "invalid memory address or nil pointer dereference" (runtime error: invalid memory address or nil pointer dereference)
/home/aledbf/go/src/github.com/kong/kubernetes-ingress-controller/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:72
/home/aledbf/go/src/github.com/kong/kubernetes-ingress-controller/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:65
/home/aledbf/go/src/github.com/kong/kubernetes-ingress-controller/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:51
/home/aledbf/.gimme/versions/go1.10.2.linux.amd64/src/runtime/asm_amd64.s:573
/home/aledbf/.gimme/versions/go1.10.2.linux.amd64/src/runtime/panic.go:502
/home/aledbf/.gimme/versions/go1.10.2.linux.amd64/src/runtime/panic.go:63
/home/aledbf/.gimme/versions/go1.10.2.linux.amd64/src/runtime/signal_unix.go:388
/home/aledbf/go/src/github.com/kong/kubernetes-ingress-controller/internal/ingress/controller/kong.go:889
/home/aledbf/go/src/github.com/kong/kubernetes-ingress-controller/internal/ingress/controller/kong.go:77
/home/aledbf/go/src/github.com/kong/kubernetes-ingress-controller/internal/ingress/controller/controller.go:128
/home/aledbf/go/src/github.com/kong/kubernetes-ingress-controller/internal/ingress/controller/run.go:85
/home/aledbf/go/src/github.com/kong/kubernetes-ingress-controller/internal/task/queue.go:112
/home/aledbf/go/src/github.com/kong/kubernetes-ingress-controller/internal/task/queue.go:59
/home/aledbf/go/src/github.com/kong/kubernetes-ingress-controller/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133
/home/aledbf/go/src/github.com/kong/kubernetes-ingress-controller/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:134
/home/aledbf/go/src/github.com/kong/kubernetes-ingress-controller/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88
/home/aledbf/go/src/github.com/kong/kubernetes-ingress-controller/internal/task/queue.go:59
/home/aledbf/.gimme/versions/go1.10.2.linux.amd64/src/runtime/asm_amd64.s:2361
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x10ec9f4]

goroutine 137 [running]:
github.com/kong/kubernetes-ingress-controller/vendor/k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	/home/aledbf/go/src/github.com/kong/kubernetes-ingress-controller/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:58 +0x107
panic(0x12a8fc0, 0x1eb6090)
	/home/aledbf/.gimme/versions/go1.10.2.linux.amd64/src/runtime/panic.go:502 +0x229
github.com/kong/kubernetes-ingress-controller/internal/ingress/controller.(*NGINXController).syncUpstreams(0xc4204c62d0, 0xc42054c2a0, 0x3, 0x4, 0xc42044c5b0, 0x2, 0x2, 0x1ecece0, 0xc400000000)
	/home/aledbf/go/src/github.com/kong/kubernetes-ingress-controller/internal/ingress/controller/kong.go:889 +0x984
github.com/kong/kubernetes-ingress-controller/internal/ingress/controller.(*NGINXController).OnUpdate(0xc4204c62d0, 0xc4203a25a0, 0x0, 0x0)
	/home/aledbf/go/src/github.com/kong/kubernetes-ingress-controller/internal/ingress/controller/kong.go:77 +0xc8
github.com/kong/kubernetes-ingress-controller/internal/ingress/controller.(*NGINXController).syncIngress(0xc4204c62d0, 0x1315960, 0xc420405180, 0xc43021bd1a, 0x61bbfaeb)
	/home/aledbf/go/src/github.com/kong/kubernetes-ingress-controller/internal/ingress/controller/controller.go:128 +0x2cd
github.com/kong/kubernetes-ingress-controller/internal/ingress/controller.(*NGINXController).(github.com/kong/kubernetes-ingress-controller/internal/ingress/controller.syncIngress)-fm(0x1315960, 0xc420405180, 0xa, 0xc4206c5d68)
	/home/aledbf/go/src/github.com/kong/kubernetes-ingress-controller/internal/ingress/controller/run.go:85 +0x3e
github.com/kong/kubernetes-ingress-controller/internal/task.(*Queue).worker(0xc4203a22d0)
	/home/aledbf/go/src/github.com/kong/kubernetes-ingress-controller/internal/task/queue.go:112 +0x34a
github.com/kong/kubernetes-ingress-controller/internal/task.(*Queue).(github.com/kong/kubernetes-ingress-controller/internal/task.worker)-fm()
	/home/aledbf/go/src/github.com/kong/kubernetes-ingress-controller/internal/task/queue.go:59 +0x2a
github.com/kong/kubernetes-ingress-controller/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1(0xc4205ecfa8)
	/home/aledbf/go/src/github.com/kong/kubernetes-ingress-controller/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133 +0x54
github.com/kong/kubernetes-ingress-controller/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc42002bfa8, 0x3b9aca00, 0x0, 0x566e535970566101, 0xc420230660)
	/home/aledbf/go/src/github.com/kong/kubernetes-ingress-controller/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:134 +0xbd
github.com/kong/kubernetes-ingress-controller/vendor/k8s.io/apimachinery/pkg/util/wait.Until(0xc4205ecfa8, 0x3b9aca00, 0xc420230660)
	/home/aledbf/go/src/github.com/kong/kubernetes-ingress-controller/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88 +0x4d
github.com/kong/kubernetes-ingress-controller/internal/task.(*Queue).Run(0xc4203a22d0, 0x3b9aca00, 0xc420230660)
	/home/aledbf/go/src/github.com/kong/kubernetes-ingress-controller/internal/task/queue.go:59 +0x55
created by github.com/kong/kubernetes-ingress-controller/internal/ingress/controller.(*NGINXController).Start
	/home/aledbf/go/src/github.com/kong/kubernetes-ingress-controller/internal/ingress/controller/run.go:152 +0xe6


```

--- a/internal/ingress/controller/kong.go
+++ b/internal/ingress/controller/kong.go
@@ -897,11 +897,26 @@ func (n *NGINXController) syncUpstreams(locations []*ingress.Location, backends 
 					if kongIngress.Upstream.Healthchecks != nil {
 						if kongIngress.Upstream.Healthchecks.Active != nil {
 							m := structs.Map(kongIngress.Upstream.Healthchecks.Active)
+							if upstream.Healthchecks == nil {
+								upstream.Healthchecks = &kongadminv1.Healthchecks{}
+							}
+							if upstream.Healthchecks.Active == nil {
+								upstream.Healthchecks.Active = &kongadminv1.ActiveHealthCheck{}
+							}
+
 							mergo.MapWithOverwrite(upstream.Healthchecks.Active, m)
 						}
 
 						if kongIngress.Upstream.Healthchecks.Passive != nil {
 							m := structs.Map(kongIngress.Upstream.Healthchecks.Passive)
+
+							if upstream.Healthchecks == nil {
+								upstream.Healthchecks = &kongadminv1.Healthchecks{}
+							}
+							if upstream.Healthchecks.Passive == nil {
+								upstream.Healthchecks.Passive = &kongadminv1.Passive{}
+							}
+
 							mergo.MapWithOverwrite(upstream.Healthchecks.Passive, m)
 						}
 					}

